src/main/java/com/ebivariation/contigalias/dus/AssemblyReportReader.java

```diff
@@ -0,0 +1,142 @@
+package com.ebivariation.contigalias.dus;
+
+import com.ebivariation.contigalias.entities.AssemblyEntity;
+import com.ebivariation.contigalias.entities.ChromosomeEntity;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.LinkedList;
+import java.util.List;
+
+public class AssemblyReportReader {
+
+    private final BufferedReader reader;
+
+    private AssemblyEntity assemblyEntity;
+
+    private boolean reportParsed = false;
+
+    public AssemblyReportReader(InputStreamReader inputStreamReader) {
+        reader = new BufferedReader(inputStreamReader);
+    }
+
+    private void parseReport() throws IOException {
```

Comment:
our code style prioritises higher level methods first, and the inner methods later. An example:

```
A() {
  B();
  D();
}
B() {
  C();
}
C() { }
D() { }
```
so please move getAssemblyEntity before parseReport.